x/marketmap/types/market.go

```diff
@@ -30,42 +28,19 @@ func (mm *MarketMap) ValidateBasic() error {
 			return fmt.Errorf("ticker string %s does not match ticker %s", tickerStr, ticker.String())
 		}
 
-		providers, ok := mm.Providers[ticker.String()]
-		if !ok {
-			return fmt.Errorf("providers for ticker %s not found", ticker.String())
-		}
-
-		if err := providers.ValidateBasic(); err != nil {
-			return err
-		}
-
 		seenCPs[ticker.String()] = struct{}{}
 	}
 
-	for tickerStr, paths := range mm.Paths {
-		cp, err := types.CurrencyPairFromString(tickerStr)
-		if err != nil {
-			return err
+	// check if all providers refer to tickers
+	for tickerStr, providers := range mm.Providers {
+		// check if the ticker is supported
+		if _, ok := mm.Tickers[tickerStr]; !ok {
+			return fmt.Errorf("provider %s refers to an unsupported ticker", tickerStr)
 		}
 
-		if err := paths.ValidateBasic(cp); err != nil {
```

Comment:
What is there reason for removing most of this logic?

Reply:
yea, we can have USDC/USDT included in a path for USDT/USD. I think we should move this logic into each aggregator that might use the paths field in unique ways.

Reply:
I see so they cyclic stuff is all gone?